include/crm/common/internal.h

```diff
@@ -223,6 +225,32 @@ gboolean pcmk__str_in_list(GList *lst, const gchar *s);
 bool pcmk__str_any_of(const char *s, ...) G_GNUC_NULL_TERMINATED;
 bool pcmk__str_none_of(const char *s, ...) G_GNUC_NULL_TERMINATED;
 
+/*!
+ * \internal
+ * \brief Parse a double-precision floating-point value from a string
+ *
+ * \param[in]      text         The string to parse
+ * \param[in]      default_text Default string to parse if \p text is
+ *                              \c NULL
+ * \param[out]     rc           Standard Pacemaker return code
+ *
+ * \return Parsed value on success, or -1 on error
+ * \note \p rc is set to \c pcmk_rc_no_output on \c NULL \p result,
+ *       \c pcmk_rc_no_input on no input string,
+ *       \c pcmk_rc_unknown_format on failed string conversion due to
+ *       invalid input, \c pcmk_rc_after_range on arithmetic overflow,
+ *       \c pcmk_rc_before_range on arithmetic underflow, or \c errno
+ *       from \c strtod() on other parse errors).
+ */
+static inline double
+pcmk__parse_double(const char *text, const char *default_text, int *rc)
```

Comment:
Wrapper function for pcmk__scan_double(). The parsed value is usually a lot more interesting to the caller than the rc, so we return the value here. This also passes NULL for end_text.

This is now internal, per your comment in the last push.